free_fleet/free_fleet/utils.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from free_fleet.ros2_types import (
+    ActionMsgs_CancelGoal_Request,
+    ActionMsgs_GoalInfo,
+    Time,
+    UUID
+)
+
+
+def namespacify(base_name: str, namespace: str, delimiter: str = '/') -> str:
+    """
+    Namespaces a base_name with namespace and delimiter.
+
+    If no namespace is provided, returns the base_name. Otherwise, naively
+    prefixes the base_name with namespace and delimiter, returning the result.
+    """
+    return f'{namespace}{delimiter}{base_name}' if len(namespace) != 0 \
```

Comment:
I wonder if the condition should be `if not namespace` which I believe would catch both empty string and `None` values.

Reply:
Yeah that'll catch more issues for sure, https://github.com/open-rmf/free_fleet/pull/145/commits/cccb4033c024b5035a97c7cda146f70970b24baf